src/network/address.rs

```diff
@@ -72,9 +72,9 @@ impl Address {
 fn addr_to_be(addr: [u16; 8]) -> [u16; 8] {
     // consensus_encode always encodes in LE, and we want to encode in BE.
     // this utility fn swap bytes before encoding so that the encoded result will be BE
-    let mut result = addr.clone();
-    for i in 0..8 {
-        result[i] = result[i].swap_bytes();
+    let mut result = addr;
+    for word in &mut result {
+        *word = word.swap_bytes();
     }
     result
```

Comment:
We don't actually need the copy here since the function takes ownership of `addr`. Also (and this is a bit tongue in cheek), words haven't been 2 bytes for ages ;) If single character local variable names are acceptable this could be written as:
```
// `consensus_encode` always encodes in LE, and we want to encode in BE. This
// utility fn swaps bytes before encoding so that the encoded result will be BE.
fn addr_to_be(mut addr: [u16; 8]) -> [u16; 8] {
    for u in &mut addr {
        *u = u.swap_bytes();
    }
    addr
}
```

Reply:
TBH, wanted to keep structure and not mess with diff too much but willing to change it.

AFAIK "word" is loosely defined. I hate single-letter variables except for `i` and `j` in loops and `x`, `y`, `z` in coordinates.

Reply:
All fair points, thanks.